web_poet/_base.py

```diff
@@ -43,8 +45,22 @@ def __init__(self, url: Union[str, '_Url']):
                             f"got {url.__class__} instance instead")
         self._url = str(url)
 
+    def join(self: T_url, other: Union[str, '_Url']) -> T_url:
+        return self.__class__(urljoin(self._url, str(other)))
+
+    def update_query(self: T_url,
+                     new_parameters: Dict[str, str]) -> T_url:
+        new_url = add_or_replace_parameters(self._url,
```

Comment:
yarl's update_query looks way more powerful than w3lib's add_or_replace_parameters, see https://yarl.readthedocs.io/en/latest/api.html#yarl.URL.update_query for the supported arguments. I like that multiple values are handled in yarl, and duplicate query parameters in general.